pkg/cedar/core/client.go

```diff
@@ -48,17 +43,43 @@ func (t *loggingTransport) RoundTrip(r *http.Request) (*http.Response, error) {
 	return resp, err
 }
 
-// NewClient builds the type that holds a connection to the CEDAR Core API
-func NewClient(ctx context.Context, cedarHost string, cedarAPIKey string, cedarAPIVersion string, cacheRefreshTime time.Duration, mockEnabled bool) *Client {
-	c := cache.New(cache.NoExpiration, cache.NoExpiration) // Don't expire data _or_ clean it up
+var cedarPath string
+
+func PurgeCacheByPath(ctx context.Context, path string) error {
+	req, err := http.NewRequest("PURGE", cedarPath+path, nil)
+	logger := appcontext.ZLogger(ctx)
+	if err != nil {
+		return err
+	}
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	var cacheIsPurged bool
+	if res.StatusCode == 200 {
+		cacheIsPurged = true
+	}
+	logger.Info(
+		"Cache Purge",
+		zap.String("pathArg", path),
+		zap.String("path", req.URL.Path),
+		zap.String("queryParams", req.URL.RawQuery),
+		zap.Int("status", res.StatusCode),
+		zap.Bool("success", cacheIsPurged),
+	)
+	return nil
+}
 
+// NewClient builds the type that holds a connection to the CEDAR Core API
+func NewClient(ctx context.Context, cedarHost string, cedarAPIKey string, cedarAPIVersion string, mockEnabled bool) *Client {
```

Comment:
could we do this via a once func and initialize `cedarPath` inside? then `cedarPath` can never change

```go
var (
	cedarPath  string
	client     *Client
	clientOnce sync.Once
)

// NewClient builds the type that holds a connection to the CEDAR Core API
func NewClient(ctx context.Context, cedarHost string, cedarAPIKey string, cedarAPIVersion string, mockEnabled bool) *Client {
	clientOnce.Do(func() {
		hc := http.Client{
			Transport: &loggingTransport{
				logger: appcontext.ZLogger(ctx),
			},
		}

		basePath := "/gateway/CEDAR Core API/" + cedarAPIVersion
		cedarPath = "http://" + cedarHost + basePath
		client = &Client{
			mockEnabled: mockEnabled,
			auth: httptransport.APIKeyAuth(
				"x-Gateway-APIKey",
				"header",
				cedarAPIKey,
			),
			sdk: apiclient.New(
				httptransport.New(
					cedarHost,
					basePath,
					[]string{"http"},
					// apiclient.DefaultSchemes,
				),
				strfmt.Default,
			),
			hc: &hc,
		}
	})

	return client
}

```

Reply:
Added in [32f011a](https://github.com/CMSgov/easi-app/pull/2545/commits/32f011ab43d228e4df364ccc7185873eff1a5363)